src/compiler/transformers/static-to-meta/component.ts

```diff
@@ -176,6 +178,42 @@ export const parseStaticComponentMeta = (
   return cmpNode;
 };
 
+const validateComponentMembers = (node: ts.Node) => {
+  /**
+   * validate if:
+   */
+  if (
+    /**
+     * the component has a getter called "shadowRoot"
+     */
+    ts.isGetAccessorDeclaration(node) &&
+    ts.isIdentifier(node.name) &&
+    node.name.escapedText === 'shadowRoot' &&
+    /**
+     * the parent node is a class declaration
+     */
+    ts.isClassDeclaration(node.parent)
+  ) {
+    const decorator = ts.getDecorators(node.parent)[0];
+    /**
+     * the class is actually a Stencil component, has a decorator with a property named "tag"
+     */
+    if (
+      ts.isCallExpression(decorator.expression) &&
+      decorator.expression.arguments.length === 1 &&
+      ts.isObjectLiteralExpression(decorator.expression.arguments[0]) &&
+      decorator.expression.arguments[0].properties.some(
+        (prop) => ts.isPropertyAssignment(prop) && prop.name.getText() === 'tag',
+      )
```

Comment:
Is it easier to just check the decorator name?

Reply:
Yes, but you could potentially import it with any arbitrary name which is why I did this extra step of checking if the decorator has 1 argument which contains a `tag` property since this is the same everywhere.

Reply:
Ah yeah, good point!